arcgis-ios-sdk-samples/Layers/Map image layer tables/MapImageLayerTablesViewController.swift

```diff
@@ -0,0 +1,158 @@
+        return renderer
+    }
+    
+    /// Query features on the first table in the map image layer.
+    func queryFeatures() {
+        // Create query parameters and set its where clause.
+        let nullCommentsParameters = AGSQueryParameters()
+        nullCommentsParameters.whereClause = "requestid <> '' AND comments <> ''"
+        // Set the first table from the map image layer.
+        commentsTable = serviceRequestsMapImageLayer.tables.first
+        // Query features on the feature table with the query parameters and all feature fields.
+        commentsTable?.queryFeatures(with: nullCommentsParameters, queryFeatureFields: .loadAll) { [weak self] result, error in
+            guard let self = self else { return }
+            if let comments = result?.featureEnumerator().allObjects {
+                // Show the records from the service request comments table in the list view control.
+                self.commentsArray = comments
+                // Enable the button after the map and features have been loaded.
+                self.queryBarButtonItem.isEnabled = true
+            } else if let error = error {
+                self.presentAlert(error: error)
+            }
+        }
+    }
+    
+    /// Query related features for the selected feature.
+    func queryCommentsTable(feature: AGSArcGISFeature) {
+        // Get the relationship that defines related service request features for features in the comments table (this is the first and only relationship).
+        let commentsRelationshipInfo = commentsTable?.layerInfo?.relationshipInfos.first
+        // Create query parameters to get the related service request for features in the comments table.
+        let relatedQueryParameters = AGSRelatedQueryParameters(relationshipInfo: commentsRelationshipInfo!)
+        relatedQueryParameters.returnGeometry = true
+        // Query related features for the selected comment and its related query parameters.
+        commentsTable?.queryRelatedFeatures(for: feature, parameters: relatedQueryParameters) { [weak self] results, error in
+            guard let self = self else { return }
+            // Get the first related feature.
+            if let relatedFeature = results?.first?.featureEnumerator().nextObject() as? AGSArcGISFeature {
+                // Load the feature and get its geometry to show as a graphic on the map.
+                relatedFeature.load { error in
+                    if let error = error {
+                        self.presentAlert(error: error)
+                    } else {
+                        // Get the feature's geometry.
+                        guard let serviceRequestPoint = relatedFeature.geometry as? AGSPoint else { return }
+                        // Create a graphic to add to the graphics overlay.
+                        let graphic = AGSGraphic()
+                        graphic.geometry = serviceRequestPoint
+                        self.selectedFeaturesOverlay.graphics.add(graphic)
```

Comment:
This workflow actually makes me wonder: is there a reason why we use `MapServer` over `FeatureServer` for this sample? Because the feature service https://sampleserver6.arcgisonline.com/arcgis/rest/services/ServiceRequest/FeatureServer is also present. Using feature service/feature layer makes it easier to show the select feature UI, by simply calling `AGSFeatureLayer.select(_:)`.

Maybe it is by design, so I don't have problem if we keep the current workflow.

Reply:
I'm also unsure about this. I'll bring this topic to the common design PR.

Reply:
Nevermind, I tried to use the feature service and it doesn't seem to work under the current workflow, showing duplicate comments.

![dup](https://user-images.githubusercontent.com/9660181/160506394-16de6ea4-88ab-41e1-b7a4-5046d8832c12.png)